src/db_state.rs

```diff
@@ -327,6 +333,12 @@ impl DbState {
         state.core.compacted = compacted;
         self.update_state(state);
     }
+
+    pub fn set_error(&mut self, error: SlateDBError) {
```

Comment:
I think we should only set the error if `state.error` is `None`. That way we get the _first_ error that occurs, which is likely to be the one we care about. Subsequent errors might be misleading since the could be a result of the first.